datafusion/physical-expr/src/expressions/is_null.rs

```diff
@@ -100,6 +110,49 @@ impl PhysicalExpr for IsNullExpr {
     }
 }
 
+pub(crate) fn union_is_null(union_array: &UnionArray) -> Result<BooleanArray> {
+    if let Some(offsets) = union_array.offsets() {
+        dense_union_is_null(union_array, offsets)
+    } else {
+        sparce_union_is_null(union_array)
+    }
+}
+
+fn dense_union_is_null(
```

Comment:
It's worth someone else stepping through this logic and checking it's right. I'm not 100% sure it's all correct.

Reply:
I did and it looks good to me